src/web/tools/KeyEventManager.ts

```diff
@@ -0,0 +1,103 @@
+import { AdaptedEvent, EventTypes } from '../interfaces';
+import EventManager from './EventManager';
+import { PointerType } from '../../PointerType';
+import { View } from 'react-native';
+
+export default class KeyEventManager extends EventManager<HTMLElement> {
+  private activationKeys = ['Enter', 'Space', 'NumpadEnter'];
+  private cancelationKeys = ['Tab'];
+
+  private keyDownCallback = (event: KeyboardEvent): void => {
+    console.log(event.code);
+    if (this.cancelationKeys.indexOf(event.code) !== -1) {
+      this.dispatchEvent(event, EventTypes.CANCEL);
+    }
+
+    if (this.activationKeys.indexOf(event.code) === -1) {
+      return;
+    }
+
+    this.dispatchEvent(event, EventTypes.DOWN);
+  };
+
+  private keyUpCallback = (event: KeyboardEvent): void => {
+    if (this.activationKeys.indexOf(event.code) === -1) {
+      return;
+    }
+
+    this.dispatchEvent(event, EventTypes.UP);
+  };
+
+  private dispatchEvent(event: KeyboardEvent, eventType: EventTypes) {
+    (event.target as unknown as View)?.measure?.(
```

Comment:
We are in `web` environment, can't we use `getBoundingClientRect`?

Reply:
Yes, but please use https://github.com/software-mansion/react-native-gesture-handler/blob/acf996d4c022e1c7c911f341521cb707ad7cb24e/src/web/tools/GestureHandlerDelegate.ts#L15

it uses `getBoundingClientRect` under the hood but abstracts away the implementation.

Reply:
Hey @j-piasecki thank you for your review.

I'm not sure if it's possible to use `measureView` inside `KeyEventManager`, as `measureView` is only present on the `GestureHandlerWebDelegate`, which we don't have access to.

I also don't think it would be possible to add access to it either, without creating a circular dependency, as `GestureHandlerWebDelegate` already imports `KeyEventManager`, and to attain this method `KeyEventManager` would have to also import `GestureHandlerWebDelegate`.

Please let me know if it's really necessary to `measureView` instead of `getBoundingClientRect`

Reply:
I'd go with `getBoundingClientRect` - we already use it in [PointerEventManager](https://github.com/software-mansion/react-native-gesture-handler/blob/acf996d4c022e1c7c911f341521cb707ad7cb24e/src/web/tools/PointerEventManager.ts#L241)